src/main.rs

```diff
@@ -402,89 +457,140 @@ fn get_args(matches: &ArgMatches) -> CliArgs {
+        )
+        .subcommand(
+            gen_metadata("start_node")
+                .arg(get_map(&arg_map, DRIVER_NAME_ARG_NAME))
+                .arg(get_map(&arg_map, END_POINT_ARG_NAME))
+                // Now gen async fuse data other args
+                .arg(get_map(&arg_map, SERVER_PORT_NUM_ARG_NAME))
+                .arg(get_map(&arg_map, VOLUME_TYPE_ARG_NAME))
+                .arg(get_map(&arg_map, CACHE_CAPACITY_ARG_NAME))
+                .arg(get_map(&arg_map, VOLUME_INFO_ARG_NAME)),
+        )
+        .subcommand(
+            gen_metadata("start_scheduler_extender")
+                .arg(get_map(&arg_map, SCHEDULER_EXTENDER_PORT_ARG_NAME)),
+        )
+        .subcommand(
+            SubCommand::with_name("async_fuse")
+                .arg(get_map(&arg_map, SERVER_PORT_NUM_ARG_NAME))
+                .arg(get_map(&arg_map, VOLUME_TYPE_ARG_NAME))
+                .arg(get_map(&arg_map, CACHE_CAPACITY_ARG_NAME))
+                .arg(get_map(&arg_map, VOLUME_INFO_ARG_NAME))
+                .arg(get_map(&arg_map, NODE_ID_ARG_NAME))
+                .arg(get_map(&arg_map, NODE_IP_ARG_NAME))
+                .arg(get_map(&arg_map, MOUNT_POINT_ARG_NAME))
+                .arg(get_map(&arg_map, ETCD_ADDRESS_ARG_NAME)),
+        )
+        .get_matches();
+    matches
+}
+
 #[allow(clippy::too_many_lines)]
 fn main() -> anyhow::Result<()> {
     env_logger::init();
-    let args = parse_args();
-
-    debug!(
-        "{}={}, {}={}, {}={}, {}={}, {}={}, {}={}, {}={:?}, {}={:?}, {}={}, {}={}, {}={}, {}={}, {}={:?}",
-        END_POINT_ARG_NAME,
-        args.end_point,
-        WORKER_PORT_ARG_NAME,
-        args.worker_port,
-        NODE_ID_ARG_NAME,
-        args.node_id,
-        NODE_IP_ARG_NAME,
-        args.ip_address,
-        DRIVER_NAME_ARG_NAME,
-        args.driver_name,
-        MOUNT_POINT_ARG_NAME,
-        args.mount_dir,
-        RUN_AS_ARG_NAME,
-        args.run_as,
-        ETCD_ADDRESS_ARG_NAME,
-        args.etcd_address_vec,
-        SCHEDULER_EXTENDER_PORT_ARG_NAME,
-        args.scheduler_extender_port,
-        CACHE_CAPACITY_ARG_NAME,
-        args.cache_capacity,
-        SERVER_PORT_NUM_ARG_NAME,
-        args.server_port,
-        VOLUME_INFO_ARG_NAME,
-        args.volume_info,
-        VOLUME_TYPE_ARG_NAME,
-        args.volume_type,
-    );
-
-    let args_clone = args.clone();
-    let etcd_delegate = EtcdDelegate::new(args.etcd_address_vec)?;
-    let etcd_delegate_clone = etcd_delegate.clone();
-    let meta_data = csi::build_meta_data(
-        args.worker_port,
-        args.node_id,
-        args.ip_address,
-        args.mount_dir,
-        args.run_as,
-        etcd_delegate,
-    )?;
-    let md = Arc::new(meta_data);
-    let async_server = true;
 
-    match args.run_as {
-        RunAsRole::Controller => {
+    let matches = parse_args();
+
+    match matches.subcommand() {
+        ("start_csi_controller", Some(matches)) => {
```

Comment:
Define const str for different subcommands.

Reply:
The 'subcommand' do not allow define const str replace the first param.